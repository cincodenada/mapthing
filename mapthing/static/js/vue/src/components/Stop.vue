<script setup lang="ts">
import type Stop from '@/models/Stop'
defineProps<{
  stop: Stop,
}>()
</script>
<template>
</template>
<style scoped>
</style>

