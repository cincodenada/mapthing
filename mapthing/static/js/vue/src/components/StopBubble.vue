<script setup lang="ts">
import type Stop from '@/models/Stop'
import { Temporal } from 'temporal-polyfill'

defineProps<{
  stop: Stop,
}>()

const formatter = new Intl.DateTimeFormat("en-us", {
  year: 'numeric',
  month: 'numeric',
  day: 'numeric',
  hour: 'numeric',
  minute: 'numeric',
  hour12: false,
});

function humanize(date) {
  const ts = Temporal.PlainDateTime.from(date).toZonedDateTime(Temporal.Now.timeZoneId()).epochMilliseconds;
  const parts = formatter.formatToParts(ts);
  return [parts[4], parts[1], ...parts.slice(0,3), {value: ' '}, ...parts.slice(6)].map(p => p.value).join('')
}


</script>
<template>
<div :data-start="stop.start_time" :data-end="stop.end_time">
  <div class="stop_time">{{humanize(stop.start_time)}}</div>
  <div class="stop_container place">
      <a href="#" class="trip_stop">
        <span>{{stop.name}}</span>
      </a>
    <span class="stop_latlon">{{stop.location.latitude}}, {{stop.location.longitude}}</span>
  </div>
  <div class="stop_time">{{humanize(stop.end_time)}}</div>
</div>
</template>
<style>
.stop_time, .stop_latlon {
    font-size: 80%;
    color: darkgray;
    line-height: 2;
}

.stop_latlon {
    padding-left: 2px;
}
</style>

