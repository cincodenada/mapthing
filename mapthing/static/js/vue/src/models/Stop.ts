import type Location from '@/models/Location'

export default interface Stop {
  id: number
  start_time: number
  start_id: number
  end_time: number
  end_id: number
  location: Location
}
