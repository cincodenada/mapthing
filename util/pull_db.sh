if [[ -z ${1} ]]
then
    echo "No name specified, exiting!"
    exit 1
elif [[ -e "$1.ab" || -e "${1}.db" ]]
then
    echo "Files exist, exiting!"
    exit 2
fi

adb backup -f ${1}.ab -noapk nl.sogeti.android.gpstracker
dd if=${1}.ab bs=1 skip=24 | python -c "import zlib,sys;sys.stdout.write(zlib.decompress(sys.stdin.read()))" | tar -xvf -
mv apps/nl.sogeti.android.gpstracker/db/*.db ${1}.db
rm -r apps
