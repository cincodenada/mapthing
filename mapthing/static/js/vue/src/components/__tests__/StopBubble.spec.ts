import { describe, it, expect } from 'vitest'

import { mount } from '@vue/test-utils'
import xmlFormat from 'xml-formatter'
import { promisify } from 'util'

import StopBubble from '../StopBubble.vue'
import Stop from '@/models/Stop'
import Locations from '@/fixtures/Location'

const clean = (html) => {
  const out = xmlFormat(`<root>${html}</root>`, { collapseContent: true, indentation: '  ' })
  return out.split('\n').slice(1,-1).join('\n').replaceAll(/^  /mg,'');
}

describe('StopBubble', () => {
  it('renders properly', () => {
    const wrapper = mount(StopBubble, { props: {
      stop: {
          start_time: '2024-10-10 20:00:00',
          end_time: '2024-10-11 08:30:00',
          location: Locations.Home,
      }
    } })
    expect(clean(wrapper.html())).toMatchSnapshot()
  })
})
