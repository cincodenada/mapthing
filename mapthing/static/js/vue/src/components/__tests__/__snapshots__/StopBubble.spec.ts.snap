// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`StopBubble > renders properly 1`] = `
"<div data-start="2024-10-10 20:00:00" data-end="2024-10-11 08:30:00">
  <div class="stop_time">2024/10/10 20:00</div>
  <div class="stop_container place">
    <a href="#" class="trip_stop">
      <span></span>
    </a>
    <span class="stop_latlon">45.51234, -122.51234</span>
  </div>
  <div class="stop_time">2024/10/11 08:30</div>
</div>
"
`;
